src/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/machines/BlockPlacer.java

```diff
@@ -53,6 +57,23 @@ public AutonomousMachineHandler getItemHandler() {
 					}
 					else {
 						block.setType(e.getItem().getType());
+						if(e.getItem().hasItemMeta() && (e.getItem().getItemMeta() instanceof BlockStateMeta)) {
+							BlockState itemBlockState = ((BlockStateMeta) e.getItem().getItemMeta()).getBlockState();
+							BlockState blockState = block.getState();
+							
+							if(blockState instanceof Nameable) {
+								((Nameable) blockState).setCustomName(e.getItem().getItemMeta().getDisplayName());
+							}
+							
+							//Update block state after changing name
+							blockState.update();
```

Comment:
Shouldnt .update() be called at the end? 
Even if it doesn't affect the inventory, it is more logical to place it below.

Reply:
It does affect the inventory as the inventory from the block will be different after .update(). Thus, I have to update() first to apply the name and then edit the latest inventory.

Reply:
It's redundant updating in general considering you aren't retrieving the state again after this, you're re-using the same one.

Reply:
It will get the Inventory based on the block instead of the blockstate but I will just change it to get a new/latest block state instead to avoid confusion.